docs/buffer.md

```diff
@@ -20,29 +21,64 @@ Buffer is a [Node.js API](https://nodejs.org/dist/latest-v8.x/docs/api/buffer.ht
 to read and write binary data accurately from JavaScript. ZJS supports a minimal
 subset of this API that will be expanded as the need arises.
 
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.  Click [here](Notes_on_WebIDL.md) for an
+explanation of zephyr.js' WebIDL conventions.
+
+```javascript
+[ Constructor(Uint8Array initial_values),
+  Constructor(unsigned long size),
+  Constructor(ByteString initial_string) ]
+interface Buffer {
+    readonly attribute unsigned long length;
+    unsigned long copy(Buffer target, optional unsigned long targetStart = 0,
+                                      optional unsigned long sourceStart = 0,
+                                      optional unsigned long sourceEnd = this.length);
+    this fill((string or Buffer or long) value, optional long offset = 0,
```

Comment:
You can not add "this" in an API. and same with this.length.

Could be `Buffer fill(value, offset, end = -1)` then you enterpret -1 to measure yourself or so